logic/broadcaster.js

```diff
@@ -225,10 +225,32 @@ __private.filterQueue = function(cb) {
 			if (broadcast.options.immediate) {
 				return setImmediate(filterCb, null, false);
 			} else if (broadcast.options.data) {
-				const transaction =
-					broadcast.options.data.transaction ||
-					broadcast.options.data.signature;
-				return __private.filterTransaction(transaction, filterCb);
+				let transactionId;
+				if (broadcast.options.data.transaction) {
+					// Look for a transaction of a given "id" when broadcasting transactions
+					transactionId = broadcast.options.data.transaction.id;
+				} else if (broadcast.options.data.signature) {
+					// Look for a corresponding "transactionId" of a given signature when broadcasting signatures
+					transactionId = broadcast.options.data.signature.transactionId;
+				}
+				if (!transactionId) {
+					return setImmediate(filterCb, null, false);
```

Comment:
Can we write a comment of why/when we might arrive here? Is it possible to have anything else in the queues rather than transactions or signatures?

Reply:
I don't think it's necessary in that case. The line will be reached when during broadcast we set: `broadcast.options.data.transaction.id` or `broadcast.options.data.transaction.id` to null/undefined/0 or we will try to filter queue with different keys than `transaction` or `signature`.